tests/kernel/threads/usage/src/task.c

```diff
@@ -0,0 +1,384 @@
+}
+
+/**
+ *
+ * @brief thread2 portion of k_yield() test
+ *
+ */
+
+void thread2_yield_test(void)
+{
+	int  i;
+
+	k_sem_take(&sem_thread2, K_FOREVER);
+
+	for (i = 0; i < 5; i++) {
+		thread2_data++;
+		k_yield();
+	}
+
+	k_sem_give(&sem_thread1);
+}
+
+/**
+ *
+ * @brief Test k_yield()
+ *
+ */
+
+void thread_yield_test(void)
+{
+	int  prev_thread2_data;
+	int  i;
+
+	thread2_data = 0;
+
+	/* 1st raise the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD1_PRIO);
+	k_sem_give(&sem_thread2);
+
+	for (i = 0; i < 5; i++) {
+		prev_thread2_data = thread2_data;
+		k_yield();
+		zassert_not_equal(thread2_data, prev_thread2_data,
+				"Iter %d.  thread2_data did not change (%d)\n",
+				i + 1, thread2_data);
+	}
+
+	/* Restore thread2 priority */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO);
+
+	/* Ensure that thread2 finishes */
+	k_sem_take(&sem_thread1, K_FOREVER);
+}
+
+/**
+ *
+ * @brief thread2 portion of k_thread_suspend() and
+ *                         k_thread_resume() tests
+ *
+ */
+
+void thread2_suspend_test(void)
+{
+	thread2_data++;
+
+	k_sem_take(&sem_thread2, K_FOREVER);
+}
+
+/**
+ *
+ * @brief Test k_thread_suspend() and k_thread_resume()
+ *
+ * This test suspends thread2. Once it is suspended, the main thread
+ * (thread1) sleeps for one second.  If thread2 is truly suspended,
+ * it will not execute and modify <thread2_data>.  Once confirmed,
+ * thread2 is resumed, and the main thread sleeps once more.  If
+ * thread2 has truly resumed, it will modify <thread2_data>.
+ *
+ */
+
+void thread_suspend_test(void)
+{
+	int  prev_thread2_data;
+
+	k_thread_suspend(thread2_id);    /* Suspend thread2 */
+
+	prev_thread2_data = thread2_data;
+	k_sleep(SLEEP_TIME);
+
+	zassert_equal(thread2_data, prev_thread2_data,
+			      "thread2 did not suspend!\n");
+
+	k_thread_resume(thread2_id);
+	k_sleep(SLEEP_TIME);
+
+	zassert_not_equal(thread2_data, prev_thread2_data,
+			      "thread2 did not resume!\n");
+
+	k_sem_give(&sem_thread2);
+}
+
+/**
+ *
+ * @brief Helper thread (thread2) to test the thread APIs
+ *
+ */
+
+void thread2_entry(void)
+{
+	k_sem_take(&sem_thread2, K_FOREVER);
+	thread_macros_test(thread2_id, THREAD2_PRIO);
+
+	k_sem_give(&sem_thread1);
+
+	thread2_set_prio_test();
+
+	thread2_sleep_test();
+
+	thread2_yield_test();
+
+	thread2_suspend_test();
+}
+
+/**
+ *
+ * @brief Main thread (thread1) to test the thread APIs
+ *
+ */
+
+void thread1_entry(void)
+{
+	TC_PRINT("Testing thread_id_get() and k_thread_priority_get()\n");
+	thread_macros_test(thread1_id, THREAD1_PRIO);
+
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+
+	TC_PRINT("Testing k_thread_priority_set()\n");
+	thread_set_prio_test();
+
+	TC_PRINT("Testing k_sleep()\n");
+	thread_sleep_test();
+
+	TC_PRINT("Testing k_yield()\n");
+	thread_yield_test();
+
+	TC_PRINT("Testing k_thread_suspend() and k_thread_resume()\n");
+	thread_suspend_test();
+}
+
+/* The test to validate thread1_entry */
+void test_thread1(void)
```

Comment:
you don't need a separate test function, just move the k_thread_create logic to in one test functions as you are testing the only one thing by using two threads. so technically this is not a different test.